src/core/tsi/alts/handshaker/alts_tsi_handshaker.cc

```diff
@@ -334,6 +334,29 @@ tsi_result alts_tsi_handshaker_result_create(grpc_gcp_HandshakerResp* resp,
                                                  local_service_account);
   grpc_gcp_AltsContext_set_peer_rpc_versions(
       context, const_cast<grpc_gcp_RpcProtocolVersions*>(peer_rpc_version));
+  grpc_gcp_Identity* peer_identity = const_cast<grpc_gcp_Identity*>(identity);
+  if (peer_identity == nullptr) {
+    gpr_log(GPR_ERROR, "Null peer identity in ALTS context.");
+    return TSI_FAILED_PRECONDITION;
+  }
+  if (grpc_gcp_Identity_has_attributes(
+          const_cast<grpc_gcp_Identity*>(peer_identity))) {
```

Comment:
Isn't the const-ness of peer_identity already removed? Do we still need to do a const_cast here?

Reply:
the grpc_gcp_Identity_has_attributes method takes in a constant Identity so I had added the cast. the test still passes when I remove the cast so I don't think it is needed